internal/v2/tls_config_store/tls_config_store.go

```diff
@@ -22,53 +25,217 @@ var (
 )
 
 // GetTlsConfigurationForClient returns a tls.Config instance for use by a client application.
-func GetTlsConfigurationForClient(serverHostname string) *tls.Config {
-	// TODO(rmehta19): Call S2Av2 for certificate.
-	// TODO(rmehta19): Call remote signer library for private key.
-	cert, err := tls.X509KeyPair(clientCert, clientKey)
+func GetTlsConfigurationForClient(serverHostname string, cstream s2av2pb.S2AService_SetUpSessionClient) (*tls.Config, error) {
+	// Send request to S2Av2 for config.
+	if err := cstream.Send(&s2av2pb.SessionReq {
+		AuthenticationMechanisms: []*s2av2pb.AuthenticationMechanism {
+			{
+				// TODO(rmehta19): Populate Authentication Mechanism using tokenmanager.
+				MechanismOneof: &s2av2pb.AuthenticationMechanism_Token{"token"},
+			},
+		},
+		ReqOneof: &s2av2pb.SessionReq_GetTlsConfigurationReq {
+			&s2av2pb.GetTlsConfigurationReq {
+				ConnectionSide: commonpb.ConnectionSide_CONNECTION_SIDE_CLIENT,
+			},
+		},
+	}); err != nil {
+		return nil, err
+	}
+
+	// Get the response containing config from S2Av2.
+	resp, err := cstream.Recv()
 	if err != nil {
-		log.Fatalf("Failed to generate X509KeyPair: %v", err)
+		return nil, err
+	}
+
+	// TODO(rmehta19): Handle resp.GetStatus().
+
+	// Extract TLS configiguration from SessionResp.
+	tlsConfig := resp.GetGetTlsConfigurationResp().GetClientTlsConfiguration()
+
+	var cert tls.Certificate
+	for _, v := range tlsConfig.CertificateChain {
+		// Populate Certificates field
+		block, _ := pem.Decode([]byte(v))
+		if block == nil {
+			return nil, errors.New("certificate in CertificateChain obtained from S2Av2 is empty.")
+		}
+		x509Cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			return nil, err
+		}
+		cert.Certificate = append(cert.Certificate, x509Cert.Raw)
```

Comment:
I just realized there is a "Leaf" field in the tls.Certificate (https://pkg.go.dev/crypto/tls#Certificate). I'm adding all the certs, including the leaf to the "Certificate". I think I should also add the leaf cert to the "Leaf" field.

Reply:
Tested with echo/server, fakes2av2_server and echo/client, and it works after this change.